cmd/compareBedDistanceBasedOnName/compareBedDistanceBasedOnName.go

```diff
@@ -0,0 +1,102 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"github.com/vertgenlab/gonomics/bed"
+	"github.com/vertgenlab/gonomics/fileio"
+	"log"
+)
+
+func compareBedDistanceBasedOnName(inputBed string, genomeBed string, outBed string) {
+	out := fileio.EasyCreate(outBed)
+	input := bed.Read(inputBed)
+	genome := bed.Read(genomeBed)
+
+	genomeMap := make(map[string][]bed.Bed)
+	var i int
+
+	for i = range genome {
+		genomeMap[genome[i].Name] = append(genomeMap[genome[i].Name], genome[i])
+	}
+
+	var inputNameField []bed.Bed
+	var found bool
+	var distance int
+	var lowestDistance int
+	var err error
+	var j int
+
+	for i := 0; i < len(input); i++ {
+		lowestDistance = 0
+		inputNameField, found = genomeMap[input[i].Name]
+		if found != true {
+			log.Fatalf("Did not find %s", input[i].Name)
```

Comment:
I was going to comment that the log.Fatal here seems like an intense reaction to not finding the gene name, but then I read that you expect to find at least 1 corresponding gene name entry in bed 2 for each entry in bed 1, so it's probably a suitable application!